frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/conditions-table/conditions-table.component.ts

```diff
@@ -98,15 +98,15 @@ export class ConditionsTableComponent implements OnInit, OnDestroy {
   }
 
   handleDesignDataChanges(designData: ExperimentFactorialDesignData) {
-    if (this.experimentInfo?.partitions.length && !this.formInitialized && !this.isAnyRowRemoved) {
+    const isDesignDataValid = this.checkDesignDataValidity(designData);
+    if (this.experimentInfo?.partitions.length && !this.isAnyRowRemoved && !this.formInitialized) {
       this.handleInitializeExistingTableData();
-    } else if (!this.experimentInfo && this.formInitialized && !this.isAnyRowRemoved) {
+    } else if (!this.experimentInfo && !this.isAnyRowRemoved && (!this.formInitialized || !isDesignDataValid)) {
```

Comment:
Do we still need `isAnyRowRemoved`? I think `isDesignDataValid` can replace it. And it doesn't go stale after multiple back-and-forth edits.

Reply:
done